src/js/api.js

```diff
@@ -318,6 +318,23 @@ export function pyimport(mod_name) {
   return Module.importlib.import_module(mod_name);
 }
 
+/**
+ * Unpack an archive into a target directory.
+ *
+ * @param {ArrayBuffer} buffer The archive as an ArrayBuffer (it's also fine to pass a TypedArray).
+ * @param {string} format The format of the archive. Should be one of the formats recognized by `shutil.unpack_archive`.
+ * By default the options are 'bztar', 'gztar', 'tar', 'zip', and 'wheel'. Several synonyms are accepted for each format, e.g.,
+ * for 'gztar' any of '.gztar', '.tar.gz', '.tgz', 'tar.gz' or 'tgz' are considered to be synonyms.
+ *
+ * @param {string=} extract_dir The directory to unpack the archive into. Defaults to the working directory.
+ */
+export function unpackArchive(buffer, format, extract_dir) {
```

Comment:
How about matching parameter order of this with `FetchResponse.unpack_archive`?

https://github.com/pyodide/pyodide/blob/d6035abb888c3388106c08402627e637591f440e/src/py/pyodide/http.py#L185

And also maybe we could make `FetchResponse.unpack_archive` to use `unpack_buffer_archive`? Because there are some duplications in the unpacking code now.

Reply:
I added `*` into the argument list of `unpack_archive` and `unpack_buffer_archive` to alleviate argument order issues. I unified `unpack_archive` and `unpack_buffer_archive` as you suggested.